internal/store/elasticsearch/discovery_search_repository.go

```diff
@@ -277,3 +293,128 @@ func (repo *DiscoveryRepository) toSuggestions(response searchResponse) (results
 
 	return
 }
+
+func (repo *DiscoveryRepository) Group(ctx context.Context, cfg asset.GroupConfig) ([]asset.GroupResult, error) {
+	if len(cfg.GroupBy) == 0 || cfg.GroupBy[0] == "" {
+		err := asset.DiscoveryError{Err: fmt.Errorf("group by field cannot be empty")}
+		return nil, err
+	}
+	query, err := repo.buildGroupQuery(cfg)
+	if err != nil {
+		err = asset.DiscoveryError{Err: fmt.Errorf("error building query %w", err)}
+		return nil, err
+	}
+
+	res, err := repo.cli.client.Search(
+		repo.cli.client.Search.WithFilterPath("aggregations"),
+		repo.cli.client.Search.WithBody(query),
+		repo.cli.client.Search.WithIgnoreUnavailable(true),
+		repo.cli.client.Search.WithContext(ctx),
+	)
+
+	if err != nil {
+		err = asset.DiscoveryError{Err: fmt.Errorf("error executing group query %w", err)}
+		return nil, err
+	}
+
+	var response groupResponse
+
+	err = json.NewDecoder(res.Body).Decode(&response)
+	if err != nil {
+		fmt.Println(err.Error())
```

Comment:
can you remove this Println statement and add a logger line if necessary

Reply:
This is already getting logged in search.go#GroupAssets(). Hence removed the Print from here.